lib/matplotlib/testing/compare.py

```diff
@@ -128,6 +132,60 @@ def convert(old, new):
     return convert
 
 
+class _SVGConverter(object):
+    def __init__(self):
+        self._proc = None
+        # We cannot rely on the GC to trigger `__del__` at exit because
+        # other modules (e.g. `subprocess`) may already have their globals
+        # set to `None`, which make `proc.communicate` or `proc.terminate`
+        # fail.  By relying on `atexit` we ensure the destructor runs before
+        # `None`-setting occurs.
+        atexit.register(self.__del__)
+
+    def _read_to_terminator(self):
+        stream = iter(functools.partial(self._proc.stdout.read, 1), b"")
+        terminator = (b"\n", b">")
+        n = len(terminator)
+        its = itertools.tee(stream, n)
```

Comment:
I wrote a slightly more general version that can handle arbitrary long terminators.  I would rather not have the implementation tied to the fact that we're looking exactly for a two-character terminator.